sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -120,7 +110,25 @@ public EventData(BinaryData body) {
         this.body = Objects.requireNonNull(body, "'body' cannot be null.");
         this.context = Objects.requireNonNull(context, "'context' cannot be null.");
         this.systemProperties =  Objects.requireNonNull(systemProperties, "'systemProperties' cannot be null.");
-        this.properties = new HashMap<>();
+        this.amqpAnnotatedMessage = new AmqpAnnotatedMessage(AmqpMessageBody.fromData(body.toBytes()));
+        if (Objects.nonNull(this.systemProperties.getOffset())) {
+            amqpAnnotatedMessage.getMessageAnnotations().put(OFFSET_ANNOTATION_NAME.getValue(),
+                this.systemProperties.getOffset());
+        }
+        if (Objects.nonNull(this.systemProperties.getSequenceNumber())) {
+            amqpAnnotatedMessage.getMessageAnnotations().put(SEQUENCE_NUMBER_ANNOTATION_NAME.getValue(),
+                this.systemProperties.getSequenceNumber());
+        }
+        if (Objects.nonNull(this.systemProperties.getPartitionKey())) {
```

Comment:
In this case, the system properties would have been returned by the Event Hubs service in normal operation;  we definitely shouldn't be validating the data sent to us by the broker;  the broker owns those fields.

Reply:
@jsquire  Does this situation also apply to other properties in System Properties (for example: Offset) ?

Reply:
Yes; the `system properties` are owned by the Event Hubs broker.  We should be surfacing them without making any changes to them or applying any kind of fallback logic to populate them.   `System Properties` is intended to be just a view over the data with no logic or other intelligence at all.

Reply:
According to your instructions, fix in new version.